scanner/decisions/0014-csaf-enricher.md

```diff
@@ -0,0 +1,88 @@
+# 0014 - CSAF Enricher
+
+- **Author(s):** Ross Tannenbaum
+- **Created:** [2024-12-05 Thurs]
+
+## Status
+
+Status: Accepted
+
+## Context
+
+Prior to 4.6.0, Scanner V4 fetched vulnerability data affecting Red Hat products from
+[Red Hat's OVAL](https://security.access.redhat.com/data/oval/v2/) feeds. This data tended to take an
+advisory-centric approach. That is, when a CVE was fixed for a particular product, the CVE entry related to that product
+would be replaced with the associated advisory (RHSA/RHEA/RHBA). This made it difficult for us to provide 
+our users with a CVE-centric vulnerability model, which we'd prefer, as CVEs are rather ubiquitous.
+
+We switched to [Red Hat's VEX](https://security.access.redhat.com/data/csaf/v2/vex/) data in 4.6.0, as it is recommended
+and is CVE-centric, as desired. Using VEX data, Scanner V4 is able to provide much greater accuracy when scanning Red Hat products.
+
+However, the StackRox product, as a whole, was not ready for this sudden CVE-centric change for Red Hat data for various reasons
+which are outside the scope of this document. So, for the 4.6.0 release, we opted to continue using VEX data
+but simply change the name and link for fixed vulnerabilities affecting Red Hat products to the associated advisory,
+essentially making it look like there was no change in data sources. This change was done in [this PR](https://github.com/stackrox/stackrox/pull/13052).
+
+When testing ACS 4.6.0, we found RHSA data would be inconsistent. After further investigation, it became clear
+the culprit was the quick (and perhaps under-tested) patch done in the PR referenced above. That PR
+solely swapped the vulnerability's name and link, but kept everything else the same. This is problematic. One reason
+is stated below:
+
+* Advisories may resolve several CVEs for a particular product. So, if a package in an image is affected by more than
+  one of these CVEs which are resolved by the same advisory, then Scanner V4 would output the same advisory multiple times
+  with different descriptions, CVSS scores, severities.
+
+## Decision
+
+This document attempts to address the Red Hat advisory inconsistencies by introducing a CSAF enricher.
+
+Red Hat offers [CSAF data](https://security.access.redhat.com/data/csaf/v2/advisories/), which is advisory-centric (ie one file per advisory).
+Scanner V4 will add another enricher, `rhel-csaf`, which will enrich Vulnerability Reports with Red Hat's CSAF data.
+
+The enricher will fetch Red Hat advisories and extract data we have determined has been inconsistent:
+
+* Description
+  * The current implementation takes the description from the CVE, so if a package is affected by two different CVEs
+    associated with the same advisory, then there is a clear inconsistency.
```

Comment:
Are you deciding to leave the original description? Not sure what is the decision you're making here.

Reply:
We currently just replace the vuln's name and link with the related RHSA. The other fields stay the same as what we'd see in the VEX file for that particular CVE. So, if an image is affected by two different CVEs which are resolved by the same RHSA, we now return two entries related to the same RHSA but different descriptions. Does that clarify things?